app/validation/validator.py

```diff
@@ -277,6 +248,122 @@ def _get_answers_for_block(block_json):
+        defined_maximum = answer.get('max_value')
+        maximum_values = self.get_defined_numeric_value(defined_maximum, MAX_NUMBER, answer_ranges)
+        maximum_values = self._convert_numeric_values_to_exclusive(defined_maximum, maximum_values,
+                                                                   'max', answer.get('decimal_places', 0))
+
+        return maximum_values
+
+    @staticmethod
+    def get_defined_numeric_value(defined_value, system_default, answer_ranges):
+        values = None
+
+        if defined_value is None:
+            values = [system_default]
+        elif 'value' in defined_value:
+            values = [defined_value.get('value')]
+        elif 'answer_id' in defined_value:
+            referred_answer = answer_ranges.get(defined_value['answer_id'])
+            if referred_answer is None:
+                values = None  # Referred answer is not  valid (picked up by _validate_referred_numeric_answer)
+            else:
+                values = referred_answer['min'] + referred_answer['max']
+
+        return values
+
+    @staticmethod
+    def _convert_numeric_values_to_exclusive(defined_value, values, min_or_max, decimal_places):
+        exclusive_values = values
+        if defined_value and defined_value.get('exclusive') and values:
+            exclusive_values = []
+            for value in values:
+                if min_or_max == 'min':
+                    exclusive_values.append(value + (1 / 10 ** decimal_places))
+                else:
+                    exclusive_values.append(value - (1 / 10 ** decimal_places))
+
+        return exclusive_values
+
+
+    def _validate_referred_numeric_answer(self, answer, answer_ranges):
+        # Referred will only be in answer_ranges if it's of a numeric type and appears earlier in the schema
+        # If either of the above is true then it will not have been given a value by _get_numeric_range_values
+        errors = []
+        if answer_ranges[answer.get('id')]['min'] is None:
+            error_message = 'The referenced answer "{}" can not be used to set the minimum of answer "{}"'\
+                .format(answer['min_value']['answer_id'], answer['id'])
+            errors.append(self._error_message(error_message))
+        if answer_ranges[answer.get('id')]['max'] is None:
+            error_message = 'The referenced answer "{}" can not be used to set the maximum of answer "{}"'\
+                .format(answer['max_value']['answer_id'], answer['id'])
+            errors.append(self._error_message(error_message))
+
+        return errors
+
+    def _validate_numeric_range(self, answer, answer_ranges):
+        errors = []
+        for max_value in answer_ranges[answer.get('id')]['max']:
+            for min_value in answer_ranges[answer.get('id')]['min']:
+                if max_value - min_value < 0:
+                    error_message = 'Invalid range of min = {} and max = {} is possible for answer "{}".'\
```

Comment:
Would this not read better as `Invalid range: The answer "{}" must be between the min = {} and max = {}`